File: src/IsbInterop.CSharp/Tasks/Wrappers/ControlJob.cs
using IsbInterop.Base.Wrappers;

namespace IsbInterop.Tasks.Wrappers
{
  /// <summary>
  /// Обертка над IControlJob.
  /// </summary>
  internal class ControlJob : CustomJob<IControlJobInfo>, IControlJob
  {
    /// <summary>
    /// Получает информацию об объекте.
    /// </summary>
    /// <returns>Информация об объекте.</returns>
    public override IControlJobInfo GetInfo()
    {
      var rcwIControlJobInfo = GetRcwObjectInfo();
      return new ControlJobInfo(rcwIControlJobInfo, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIControlJob">СOM-объект IControlJob.</param>
    /// <param name="scope">Область видимости.</param>
    public ControlJob(object rcwIControlJob, IScope scope)
      : base(rcwIControlJob, scope) { }
  }
}

File: src/IsbInterop/Tasks/Wrappers/ControlJob.vb
Imports IsbInterop.Base.Wrappers

Namespace Tasks.Wrappers

  ''' <summary>
  ''' Обертка над IControlJob
  ''' </summary>
  Friend Class ControlJob
    Inherits CustomJob(Of IControlJobInfo)
    Implements IControlJob

    ''' <summary>
    ''' Получает информацию об объекте.
    ''' </summary>
    ''' <returns>Информация об объекте.</returns>
    Public Overrides Function GetInfo() As IControlJobInfo
      Dim rcwIControlJobInfo = GetRcwObjectInfo()
      Return New ControlJobInfo(rcwIControlJobInfo, Scope)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIControlJob">СOM-объект IControlJob.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIControlJob As Object, scope As IScope)
      MyBase.New(rcwIControlJob, scope)
    End Sub
  End Class
End Namespace